File: KnowledgeCenterServer/_CapLab/KnowledgeCenter.CapLab.Contracts/Project.cs
using System;
using System.Collections.Generic;

namespace KnowledgeCenter.CapLab.Contracts
{
    public class Project
    {
        public int Id { get; set; }
        public string Title { get; set; }
        public string ShortDescription { get; set; }
        public string Description { get; set; }
        public string Image { get; set; }
        public DateTime CreationDate { get; set; }
        public DateTime ModificationDate { get; set; }

        public User User { get; set; }
        public ProjectStatus Status { get; set; }
        public List<Tag> Tags { get; set; }

        public int LikeCount { get; set; }
        public decimal LikeAverageRate { get; set; }
        public ProjectLike MyLike { get; set; }
        public string ApproverFullName { get; set; }
    }
}

File: KnowledgeCenterClient/src/app/caplab/models/Project.ts
import { User } from 'src/app/shared/models/User';
import { SimpleEnum } from 'src/app/shared/models/SimpleEnum';

export interface Project {
  id: number;
  title: string;
  shortDescription: string;
  description: string;
  image: string;
  creationDate: Date;
  modificationDate: Date;
  likeCount: number;
  likeAverageRate: number;
  user: User;
  status: SimpleEnum;
  tags: Array<SimpleEnum>;
  myLike: ProjectLike;
  approverFullName: string;
}

export interface ProjectLike {
  id: number;
  rate: number;
  dateCreation: Date;
}

export interface ProjectFilters {
  isOnlyMine: boolean;
  keyword: string;
  statusCodes: Array<ProjectStatus>;
  orderByDescendingCreationDate: boolean;
}

export enum ProjectStatus {
  WAITING = 'WAITING',
  VALIDATED = 'VALIDATED',
  REJECTED = 'REJECTED',
  INPROGRESS = 'INPROGRESS',
  FINISHED = 'FINISHED'
}

export enum ProjectType {
  MY,
  PROJECTS,
  ADMIN,
  INPROGRESS,
  FINISHED
}
export interface ProjectTags {
  id: number;
  projectId: number;
  tagId: number;
}

